### DISCUSSION TITLE
Why illa always use previously configed ip address to connect api port 9999

### DISCUSSION THREAD

**Discussion by TheFairyTale** [illacloud/illa-builder#542] [Q&A]
firstly, i config ip address use `illa deploy --self --port=81 --server-addr=0.0.0.0`
then i open illa use host ip address to login, the page try to connect `0.0.0.0:9999`. 
now i removed previous illa, then use `illa deploy --self --port=81 --server-addr=124.70.1.1`, i open login page again but still receive same error.
![request still try to connect 0.0.0.0:9999](https://user-images.githubusercontent.com/16740355/203605064-aea5d6ee-d89a-4d74-bc8e-c35ff2f21f53.png)

**Comment by TheFairyTale** [comment]
oh, this is my mistake, when i use different browser it‘s working fine.😊

**Reply by sjj2001** [reply]
网络异常，i can not sign in  and get http://localhost:9999/api/v1/auth/signin error

**Reply by fengdu126** [reply]
私有化部署后，我也遇到网络异常的问题，不能注册用户

**Comment by karminski** [comment]
@sjj2001 @fengdu126 

We released the version 1.4.0. it include the bug fix.

Please pull the latest docker image and try out.

And if deploy on server/virtual machine, don't forget set the environment variables in scripts/run.sh:

# builder-backend api server serve address
API_SERVER_ADDRESS=your_server_ip_address

# builder-backend websocket server serve address
WEBSOCKET_SERVER_ADDRESS=your_server_ip_address
For more details, please check out the [README.md](https://github.com/illacloud/deploy-illa-manually/blob/main/docker/README.md).
